packages/assets-controllers/src/TokenBalancesController.ts

```diff
@@ -107,526 +98,405 @@ export type AllowedEvents =
+    ];
+
     this.setIntervalLength(interval);
 
-    // Set initial preference for querying multiple accounts, and subscribe to changes
-    this.#queryMultipleAccounts = this.#calculateQueryMultipleAccounts(
-      this.messagingSystem.call('PreferencesController:getState'),
-    );
-    this.messagingSystem.subscribe(
-      'PreferencesController:stateChange',
-      this.#onPreferencesStateChange.bind(this),
+    // initial token state & subscriptions
+    const { allTokens, allDetectedTokens } = this.messagingSystem.call(
+      'TokensController:getState',
     );
-
-    // Set initial tokens, and subscribe to changes
-    ({
-      allTokens: this.#allTokens,
-      allDetectedTokens: this.#allDetectedTokens,
-    } = this.messagingSystem.call('TokensController:getState'));
+    this.#allTokens = allTokens;
+    this.#detectedTokens = allDetectedTokens;
 
     this.messagingSystem.subscribe(
       'TokensController:stateChange',
-      this.#onTokensStateChange.bind(this),
+      this.#onTokensChanged,
     );
-
-    // Subscribe to network state changes
     this.messagingSystem.subscribe(
       'NetworkController:stateChange',
-      this.#onNetworkStateChange.bind(this),
+      this.#onNetworkChanged,
     );
-
-    // subscribe to account removed event to cleanup stale balances
-
     this.messagingSystem.subscribe(
       'KeyringController:accountRemoved',
-      (accountAddress: string) => this.#handleOnAccountRemoved(accountAddress),
+      this.#onAccountRemoved,
     );
   }
 
-  /**
-   * Determines whether to query all accounts, or just the selected account.
-   * @param preferences - The preferences state.
-   * @param preferences.isMultiAccountBalancesEnabled - whether to query all accounts (mobile).
-   * @param preferences.useMultiAccountBalanceChecker - whether to query all accounts (extension).
-   * @returns true if all accounts should be queried.
-   */
-  #calculateQueryMultipleAccounts = ({
-    isMultiAccountBalancesEnabled,
-    useMultiAccountBalanceChecker,
-  }: PreferencesState & { useMultiAccountBalanceChecker?: boolean }) => {
-    return Boolean(
-      // Note: These settings have different names on extension vs mobile
-      isMultiAccountBalancesEnabled || useMultiAccountBalanceChecker,
-    );
-  };
-
-  /**
-   * Handles the event for preferences state changes.
-   * @param preferences - The preferences state.
-   */
-  #onPreferencesStateChange = (preferences: PreferencesState) => {
-    // Update the user preference for whether to query multiple accounts.
-    const queryMultipleAccounts =
-      this.#calculateQueryMultipleAccounts(preferences);
-
-    // Refresh when flipped off -> on
-    const refresh = queryMultipleAccounts && !this.#queryMultipleAccounts;
-    this.#queryMultipleAccounts = queryMultipleAccounts;
-
-    if (refresh) {
-      this.updateBalances().catch(console.error);
-    }
-  };
+  // ───────── private helpers ─────────
+  get #chainIdsWithTokens(): ChainIdHex[] {
```

Comment:
nit - hmm I think I prefer a method over a `get`, esp since this returns a new array on call.

E.g.
```ts
// Getter
const targetChains = chainIds ?? this.#chainIdsWithTokens;

// Method
const targetChains = chainIds ?? this.#chainIdsWithTokens();
```